### ISSUE TITLE
New Pipe Sponsor Block isn't responding! Even though the app is working fine this dialogue box keeps appearing, forcing me to close the app. Please check on this. My phone is Moto G73 5g. Thank you :)

### ISSUE THREAD

**Issue by Ksanbah** [polymorphicshade/NewPipe#330]
### Checklist

- [X] I am able to reproduce the bug with the [latest version](https://github.com/polymorphicshade/NewPipe/releases/latest).
- [X] I am *not* able to reproduce the bug with the *same version* of [upstream NewPipe](https://github.com/TeamNewPipe/NewPipe/releases/).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/polymorphicshade/NewPipe/issues) or [closed](https://github.com/polymorphicshade/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/polymorphicshade/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.25.1

### Steps to reproduce the bug

1. Open app
2. Play a video
3. Wait for "NewPipe SponsorBlock isn't responding" with options "close app" and "wait" to appear for no obvious reason. 

### Expected behavior

The video should just play normally. 

### Actual behavior

This pop-up box appears and ruins the experience no matter how many times you tap "wait" it keeps appearing, forcing you to tap "close app", closing the app. 

### Logs

![Screenshot_20230514-072357](https://github.com/polymorphicshade/NewPipe/assets/51513596/56cf9607-f9f4-4dae-9811-292c8604c4aa)

### Affected Android/Custom ROM version

Android 13

### Affected device model

Motorola G73 5g

**Comment by Atemu** [comment]
How does upstream NewPipe behave?

**Comment by Ksanbah** [comment]
As normal. This strange bug only occurs in your Fork.

**Comment by anonymousx97** [comment]
yes newpipe sponsorblock version (upstream is fine.) keeps going into ANR state when you have an instance open and you play another video on top of it...

steps to reproduce:
open newpipe and play something.
now play something else / share link to media player.
once it starts playing the new video, it should go into infinite ANR state even though app is working fine.

screen recording of the bug:
https://graph.org/newpip-bug-05-26
video is sped up due to file size restrictions.

**Comment by Atemu** [comment]
Given that I am *not* experiencing this issue, we'll need to figure out what the differences between our setups are.

Please provide more detail about:

- Operating system
- Device
- System settings (dev settings: show background ANRs?)
- App settings

Back up your settings etc. via the export function and clear the app data. Does the bug still occur with default settings? If not, what do you need to turn on for the bug to occur again?

**Comment by anonymousx97** [comment]
OS: Pixel experience Plus - Android 13
Device: Redmi note 8 pro
Developer options is disabled.

I cleared newpipe data after exporting my settings and while trying to recreate previous bug 
I had this crash:
https://pastebin.com/9PDnSjWu 
i played 3-4 videos by sharing links to media player and newpipe stacked them on top of one another, when i pressed back it tried to navigate back to previously played video and crashed. It always is crashing while going back to previously played video in current session.

I was able to reproduce the ANR bug using the steps mentioned in earlier comment using these settings
see if it behaves the same for you after importing them.
https://oshi.at/jqWM

it also crashed once again after going into ANR, the log was same as the one i linked earlier.

ps: I had to play 4-5 videos back to back to reproduce ANR bug this time (prolly coz I'm playing songs and not full videos) but it usually happens on 2nd video played while watching longer videos, I've faced it many times where after I'm done watching something and i play another it goes into ANR.

**Comment by Atemu** [comment]
@anonymousx97 could you provide reproduction steps from an empty config/fresh install to ANR?

A video would be fine; just upload here right on Github at full length and regular speed. It's hard to tell what's going on in your original video.

**Comment by anonymousx97** [comment]
@Atemu 
I was not able to upload the video here so uploaded to:
https://oshi.at/cjHo

I managed to record 2 crashes and 1 ANR in the video.
hope that's enough to reproduce it on your end.

**Comment by Atemu** [comment]
Could you try to repro with default settings? (Not even sponsorblock.)

**Comment by anonymousx97** [comment]
@Atemu hi!
sorry for delayed the response;
i turned off sponsorblock while keeping other settings in the video as is and I got ANR once i finished watching a video and then played another one.

then

i clear newpipe's data and didn't change any settings and started using it as i would normally use like mentioned earlier i.e play videos one after another.
after watching more than 3-4 videos it did not go into ANR state.
So either one of the settings i turned on is causing the ANR or sponsorblock is.
i think its the start video in full screen and pause video on minimise setting might be causing the ANRs.

**Comment by Atemu** [comment]
I suspected something like that might be the case. Turn on settings one-by-one to find out which one causes the issue.

Then also try turning that same set of settings on in regular upstream to see whether this is actually a bug on our side or not.

**Comment by anonymousx97** [comment]
@Atemu 
upstream doesn't have these issues.
i cleared all data of sponsorblock version and set minimise player action to None, it started giving ANR on app switch and crashed 2 times.
logs are the same ones i submitted earlier.

**Comment by YourFin** [comment]
I can confirm that this is an upstream issue. I've been having the same symptoms for a long time, and only just discovered this fork.
